lab5/DescendentRecursiveParser.cpp

```diff
@@ -0,0 +1,148 @@
+//
+// Created by radoo on 12.12.2022.
+//
+
+#include "DescendentRecursiveParser.h"
+
+void DescendentRecursiveParser::expand(Configuration &config, Grammar grammar,
+                                       const std::vector<std::string> &input) {
+    if (!grammar.isTerminal(config.inputStack.top())) {
```

Comment:
Maybe it wouldn't hurt a lot to check if the config.inputStack is empty before making the call to top()